### ISSUE TITLE
the server has asked for the client to provide credentials

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#13067]
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_gce/870/testReport/junit/(root)/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/

```
/tmp/openshift/tito/rpmbuild-originbVKDf0/BUILD/origin-1.5.0/_output/local/go/src/github.com/openshift/origin/test/extended/builds/new_app.go:32
Expected error:
    <*errors.errorString | 0xc4212baa70>: {
        s: "timed out waiting for deployment \"a234567890123456789012345678901234567890123456789012345678\" after 15m0s",
    }
    timed out waiting for deployment "a234567890123456789012345678901234567890123456789012345678" after 15m0s
not to have occurred
/tmp/openshift/tito/rpmbuild-originbVKDf0/BUILD/origin-1.5.0/_output/local/go/src/github.com/openshift/origin/test/extended/builds/new_app.go:31
```

```
18:00:05 -0500 EST  } {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-02-22 17:59:35 -0500 EST  }]
Feb 22 18:18:31.243: INFO: a234567890123456789012345678901234567890123456789012345678-1-build  ci-prtest870-ig-n-r0wr  Failed          [{Initialized True 0001-01-01 00:00:00 +0000 UTC 2017-02-22 18:03:31 -0500 EST  } {Ready False 0001-01-01 00:00:00 +0000 UTC 2017-02-22 18:04:05 -0500 EST ContainersNotReady containers with unready status: [sti-build]} {PodScheduled True 0001-01-01 00:00:00 +0000 UTC 2017-02-22 18:03:31 -0500 EST  }]
```

Looks like the build failed, you'll need more debugging.  @bparees

**Comment by bparees** [comment]
@jim-minter, as @smarterclayton says, we should update the test case to first check for build success (and dump logs on failure) before waiting for the deployment.

@smarterclayton what's the implication of
```
Feb 22 18:18:31.153: INFO: At 2017-02-22 18:04:05 -0500 EST - event for a234567890123456789012345678901234567890123456789012345678-1-build: {kubelet ci-prtest870-ig-n-r0wr} Killing: Killing container with docker id 3e745c4dbdb0: Need to kill pod.
```

is that a benign result of the build pod failing, or is that the reason the build pod failed (something killed it)?

**Comment by smarterclayton** [comment]
I *think* that's the kubelet doing its normal "all containers exited, clean
up".  But it could be a hung build if the build locked up (didn't look)

On Feb 22, 2017, at 8:59 PM, Ben Parees <notifications@github.com> wrote:

@jim-minter <https://github.com/jim-minter>, as @smarterclayton
<https://github.com/smarterclayton> says, we should update the test case to
first check for build success (and dump logs on failure) before waiting for
the deployment.

@smarterclayton <https://github.com/smarterclayton> what's the implication
of

Feb 22 18:18:31.153: INFO: At 2017-02-22 18:04:05 -0500 EST - event
for a234567890123456789012345678901234567890123456789012345678-1-build:
{kubelet ci-prtest870-ig-n-r0wr} Killing: Killing container with
docker id 3e745c4dbdb0: Need to kill pod.

is that a benign result of the build pod failing, or is that the reason the
build pod failed (something killed it)?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/issues/13067#issuecomment-281867445>,
or mute the thread

**Comment by jim-minter** [comment]
https://github.com/openshift/origin/pull/13074 adds additional logging as requested, but I think what would be more useful here is access to the master log and ideally node logs from the run.  These are not recovered as part of the CI job.  @stevekuznetsov, is this planned?

**Comment by stevekuznetsov** [comment]
No, I assume @smarterclayton wanted to retrieve both master and node logs. This is the one test I do not intimately own so I'll throw that one over the wall back to him :)

**Comment by jim-minter** [comment]
@stevekuznetsov I mean, IMO it would be good if GCE Jenkins jobs retrieved master and ideally node logs and stored them in the case of a run failure.  Is this planned?

**Comment by stevekuznetsov** [comment]
Yes, I agree that would be ideal. Clayton knows the ins and outs of that specific job much better than I, so I would assign an issue for it to him.

**Comment by bparees** [comment]
looks like we got good build logging from this failure here and it shows a problem pushing to the registry.. assigning to @mfojtik 
https://ci.openshift.redhat.com/jenkins/job/zz_origin_gce_image/138/testReport/junit/%28root%29/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/

```
2017-02-27T19:06:55.909760000Z Pushing image 172.30.241.235:5000/extended-test-new-app-flk70-2vm8c/a234567890123456789012345678901234567890123456789012345678:latest ...
2017-02-27T19:06:56.040451000Z Registry server Address: 
2017-02-27T19:06:56.040798000Z Registry server User Name: serviceaccount
2017-02-27T19:06:56.041079000Z Registry server Email: serviceaccount@example.org
2017-02-27T19:06:56.041370000Z Registry server Password: <<non-empty>>
2017-02-27T19:06:56.041669000Z error: build error: Failed to push image: Get https://172.30.241.235:5000/v1/_ping: x509: cannot validate certificate for 172.30.241.235 because it doesn't contain any IP SANs
```

**Comment by smarterclayton** [comment]
This is https://bugzilla.redhat.com/show_bug.cgi?id=1427040

**Comment by stevekuznetsov** [comment]
Saw this again [here](https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/22/testReport/junit/(root)/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/) ... not on GCE, so we have node, master logs, docker logs, whatever you need.

**Comment by jim-minter** [comment]
@mfojtik please could you take a look at this?  The build failed because the push failed part way through:

```
2017-03-11T02:50:22.815884000Z Pushed 7/9 layers, 84% complete
2017-03-11T02:50:56.735001000Z Registry server Address: 
2017-03-11T02:50:56.742434000Z Registry server User Name: serviceaccount
2017-03-11T02:50:56.747935000Z Registry server Email: serviceaccount@example.org
2017-03-11T02:50:56.757086000Z Registry server Password: <<non-empty>>
2017-03-11T02:50:56.761795000Z error: build error: Failed to push image: unauthorized: authentication required
```

```
time="2017-03-11T02:50:56.607738296Z" level=error msg="Get user failed with error: the server has asked for the client to provide credentials (get users ~)" go.version=go1.7.4 http.request.h
ost="172.30.125.183:5000" http.request.id=11b11ffe-6b7b-4a56-9585-45c526de8f78 http.request.method=PUT http.request.remoteaddr="10.128.0.1:36682" http.request.uri="/v2/extended-test-new-app-
n1n3l-3vtvl/a234567890123456789012345678901234567890123456789012345678/blobs/uploads/8a1fb4db-ceae-457a-99f5-bcf461a47676?_state=9O2ALNijpG1rx_8F8Xlx5uL8wwapUKWpnEZwrlmjLAF7Ik5hbWUiOiJleHRlb
mRlZC10ZXN0LW5ldy1hcHAtbjFuM2wtM3Z0dmwvYTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0NTY3OCIsIlVVSUQiOiI4YTFmYjRkYi1jZWFlLTQ1N2EtOTlmNS1iY2Y0NjFhNDc2NzYiLCJPZmZzZXQ
iOjcwMDY4OTM3LCJTdGFydGVkQXQiOiIyMDE3LTAzLTExVDAyOjUwOjE0WiJ9&digest=sha256%3Ad1e4228e5083cf1fb388732eea618c296623860fd76a8af1eb5f24f9e819c065" http.request.useragent="docker/1.12.6 go/go1.7
.4 kernel/3.10.0-327.22.2.el7.x86_64 os/linux arch/amd64 UpstreamClient(go-dockerclient)" instance.id=9eb7f703-fd6c-42b9-b45d-69a5196f2ad6 openshift.logger=registry vars.name="extended-test-
new-app-n1n3l-3vtvl/a234567890123456789012345678901234567890123456789012345678" vars.uuid=8a1fb4db-ceae-457a-99f5-bcf461a47676 
time="2017-03-11T02:50:56.607902636Z" level=error msg="error authorizing context: access denied" go.version=go1.7.4 http.request.host="172.30.125.183:5000" http.request.id=11b11ffe-6b7b-4a56
-9585-45c526de8f78 http.request.method=PUT http.request.remoteaddr="10.128.0.1:36682" http.request.uri="/v2/extended-test-new-app-n1n3l-3vtvl/a23456789012345678901234567890123456789012345678
9012345678/blobs/uploads/8a1fb4db-ceae-457a-99f5-bcf461a47676?_state=9O2ALNijpG1rx_8F8Xlx5uL8wwapUKWpnEZwrlmjLAF7Ik5hbWUiOiJleHRlbmRlZC10ZXN0LW5ldy1hcHAtbjFuM2wtM3Z0dmwvYTIzNDU2Nzg5MDEyMzQ1N
jc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MDEyMzQ1Njc4OTAxMjM0NTY3OCIsIlVVSUQiOiI4YTFmYjRkYi1jZWFlLTQ1N2EtOTlmNS1iY2Y0NjFhNDc2NzYiLCJPZmZzZXQiOjcwMDY4OTM3LCJTdGFydGVkQXQiOiIyMDE3LTAzLTExVDAyOjUwOjE0WiJ
9&digest=sha256%3Ad1e4228e5083cf1fb388732eea618c296623860fd76a8af1eb5f24f9e819c065" http.request.useragent="docker/1.12.6 go/go1.7.4 kernel/3.10.0-327.22.2.el7.x86_64 os/linux arch/amd64 Ups
treamClient(go-dockerclient)" instance.id=9eb7f703-fd6c-42b9-b45d-69a5196f2ad6 openshift.logger=registry vars.name="extended-test-new-app-n1n3l-3vtvl/a234567890123456789012345678901234567890
123456789012345678" vars.uuid=8a1fb4db-ceae-457a-99f5-bcf461a47676 
```

**Comment by mfojtik** [comment]
@miminar PTAL

**Comment by miminar** [comment]
@jim-minter This error is different than the one above. The last time I saw this was a [routing issue](https://github.com/openshift/origin/pull/11391) but that shouldn't be a problem now unless router is broken.

Do you have more details? Can you link the failed job?

**Comment by jim-minter** [comment]
@miminar (steve linked it above, but here you go): https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/22/testReport/junit/(root)/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/

**Comment by miminar** [comment]
@jim-minter my bad, I thought they are unrelated. Thanks.

**Comment by miminar** [comment]
@stevekuznetsov just a suggestion. I'd welcome the exact docker version in the logs. `1.12.6` is too generic (it has 13 releases in rhel7.3 already).

**Comment by stevekuznetsov** [comment]
@miminar we have an artifact called [`generated/installed_packages.log`](https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/22/s3/download/generated/installed_packages.log) which has the full list of packages installed. From that list we can see:

```
docker.x86_64                  2:1.12.6-13.el7     @httpsmirroropenshiftcomenterpriserheldockertestedx8664os
docker-client.x86_64           2:1.12.6-13.el7     @httpsmirroropenshiftcomenterpriserheldockertestedx8664os
docker-common.x86_64           2:1.12.6-13.el7     @httpsmirroropenshiftcomenterpriserheldockertestedx8664os
docker-rhel-push-plugin.x86_64 2:1.12.6-13.el7     @httpsmirroropenshiftcomenterpriserheldockertestedx8664os
```

**Comment by miminar** [comment]
@stevekuznetsov Great, thanks!

**Comment by miminar** [comment]
Possible duplicate: #12474

**Comment by miminar** [comment]
It looks like some error in master. Here's a shortened log message from the comment above: 

```
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal dockerd-current[11880]: time="2017-03-11T02:50:56.607738296Z" level=error msg="Get user failed with error: the server has asked for the client to provide credentials (get users ~)" go.version=go1.7.4 http.request.host="172.30.125.183:5000" http.request.id=11b11ffe-6b7b-4a56-9585-45c526de8f78 http.request.method=PUT http.request.remoteaddr="10.128.0.1:36682" http.request.uri="/v2/extended-test-new-app-n1n3l-3vtvl/a234567890123456789012345678901234567890123456789012345678/blobs/uploads/8a1fb4db-ceae-457a-99f5-bcf461a47676?…
```

And here are corresponging messages from master log (note the timestamp):

```
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: I0310 21:50:56.024512    2951 jwt.go:310] Could not retrieve service account extended-test-new-app-n1n3l-3vtvl/builder: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member https://ip-172-18-13-230.ec2.internal:2379 has no leader
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: I0310 21:50:56.026331    2951 jwt.go:299] Could not retrieve token openshift-infra/job-controller-token-p57nt for service account openshift-infra/job-controller: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member https://ip-172-18-13-230.ec2.internal:2379 has no leader
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: I0310 21:50:56.028512    2951 panics.go:76] GET /api/v1/namespaces/e2e-tests-prestop-5cjmf: (1.145949ms) 200 [[extended.test/v1.5.2+43a9be4 (linux/amd64) kubernetes/43a9be4] 172.18.13.230:36138]
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: E0310 21:50:56.037566    2951 errors.go:63] apiserver received an error that is not an unversioned.Status: client: etcd member https://ip-172-18-13-230.ec2.internal:2379 has no leader
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: I0310 21:50:56.037730    2951 trace.go:61] Trace "Delete /oapi/v1/namespaces/e2e-tests-prestop-5cjmf/policybindings/:default" (started 2017-03-10 21:50:46.035721639 -0500 EST):
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: [26.087µs] [26.087µs] About do delete object from database
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: [10.001988162s] [10.001962075s] END
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: I0310 21:50:56.037762    2951 panics.go:76] DELETE /oapi/v1/namespaces/e2e-tests-prestop-5cjmf/policybindings/:default: (10.002299509s) 500
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: goroutine 1391850 [running]:
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/httplog.(*respLogger).recordStatus(0xc436d081c0, 0x1f4)
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: /go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/httplog/log.go:219 +0xbb
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/httplog.(*respLogger).WriteHeader(0xc436d081c0, 0x1f4)
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: /go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/httplog/log.go:198 +0x35
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/apiserver/metrics.(*responseWriterDelegator).WriteHeader(0xc435988cf0, 0x1f4)
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: /go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/apiserver/metrics/metrics.go:117 +0x45
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/apiserver.writeNegotiated(0x92e48c0, 0xc4220089c0, 0x0, 0x0, 0x423a4e5, 0x2, 0x92ddc00, 0xc438a7c318, 0xc432db8690, 0x1f4, ...)
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: /go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/apiserver/apiserver.go:463 +0x162
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/apiserver.errorNegotiated(0x92a1400, 0xc4361f2500, 0x92e48c0, 0xc4220089c0, 0x0, 0x0, 0x423a4e5, 0x2, 0x92ddc00, 0xc438a7c318, ...)
…
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: [[openshift/v1.5.0 (linux/amd64) openshift/d49d401] 172.18.13.230:57134]
Mar 10 21:50:56 ip-172-18-13-230.ec2.internal origin-master[2951]: E0310 21:50:56.038038    2951 factory.go:49] client: etcd member https://ip-172-18-13-230.ec2.internal:2379 has no leader
```

See the full log at https://ci.openshift.redhat.com/jenkins/job/merge_pull_request_openshift_ansible/22/s3/download/journals/origin-master.service

**Comment by miminar** [comment]
Not a registry bug. @mfojtik do you know a more qualified person with enough time to hunt this down?

**Comment by stevekuznetsov** [comment]
@smarterclayton didn't we fix the etcd issues with the quorum wait logic you added>

**Comment by mfojtik** [comment]
@bparees @jim-minter @miminar @stevekuznetsov so there were 3 issues really here:

1) new-app behaving weird on 58 characters in name
2) push to registry failure (https://github.com/openshift/origin/issues/13067#issuecomment-282828731)
3) etcd quorum wait logic (https://github.com/openshift/origin/issues/13067#issuecomment-286768726) /cc @php-coder 

Do we want to open a separate issue for each and keep open for the new-app issue?

**Comment by jim-minter** [comment]
@mfojtik I think 1) happened because of 2), which happened because of 3).  I think `etcd member https://ip-172-18-13-230.ec2.internal:2379 has no leader` is the problem, whether it's dealt with in this issue or another.

**Comment by smarterclayton** [comment]
Was this in the middle of a run?  At the beginning?

**Comment by pweil-** [comment]
@mfojtik adding you to this since it seems to manifest in the unauthorized error from the registry which has been popping up more and more.  We need to determine if they are related (reference the email from  @mpbarrett)

**Comment by bparees** [comment]
unassigning myself, there's no aspect of this that appears to be a build problem at this point, it's a failure to push to the registry, possibly caused by etcd issues.

**Comment by miminar** [comment]
> Was this in the middle of a run? At the beginning?

Test 191 out of 212.

Maybe it's somehow related to the preceding test:

```
   189  21:54:56  [It] should create a job from an image, then delete the job [Conformance]
   190  21:54:56  [It] should support rolling-update to same image [Conformance]
   191  21:55:26  [Fail] [builds][Conformance] oc new-app [It] should succeed with a --name of 58 characters
```

**Comment by miminar** [comment]
> @mfojtik adding you to this since it seems to manifest in the unauthorized error from the registry which has been popping up more and more. We need to determine if they are related (reference the email from @mpbarrett)

@pweil- not sure, what email do you refer to. Is there, by a chance, swift back-end involved? There's at least a workaround for unauthenticated error with this back-end: https://github.com/openshift/origin/issues/12474#issuecomment-288097828

**Comment by stevekuznetsov** [comment]
I took a peek at a recent failure ... here's what I found.

We see requests for this namespace get slow, then fail on auth:
```
Mar 27 21:33:21 origin-master[4214]: I0327 21:33:21.018223    4214 panics.go:76] GET /oapi/v1/namespaces/extended-test-new-app-vqtp3-8lw91/builds?fieldSelector=name%3Da234567890123456789012345678901234567890123456789012345678-1: (5.851736ms) 200 [[extended.test/v3.6.0 (linux/amd64) openshift/959be87] 172.18.8.128:33980]
Mar 27 21:33:44 origin-master[4214]: I0327 21:33:44.806795    4214 panics.go:76] GET /oapi/v1/namespaces/extended-test-new-app-vqtp3-8lw91/builds?fieldSelector=name%3Da234567890123456789012345678901234567890123456789012345678-1: (3.796304658s) 200 [[extended.test/v3.6.0 (linux/amd64) openshift/959be87] 172.18.8.128:33980]
Mar 27 21:33:56 origin-master[4214]: I0327 21:33:56.011534    4214 panics.go:76] GET /oapi/v1/namespaces/extended-test-new-app-vqtp3-8lw91/builds?fieldSelector=name%3Da234567890123456789012345678901234567890123456789012345678-1: (10.001040883s) 401 [[extended.test/v3.6.0 (linux/amd64) openshift/959be87] 172.18.8.128:33980]
Mar 27 21:33:56 origin-master[4214]: E0327 21:33:56.063101    4214 errors.go:63] apiserver received an error that is not an unversioned.Status: client: etcd cluster is unavailable or misconfigured; error #0: client: etcd member https://ip-172-18-8-128.ec2.internal:2379 has no leader
```

At this time, with no other previous logs in the preceding couple of minutes, `etcd` gets slow then goes unresponsive:

```
Mar 27 21:33:18 etcd[25080]: sync duration of 1.763586557s, expected less than 1s
Mar 27 21:33:29 etcd[25080]: sync duration of 5.424003686s, expected less than 1s
Mar 27 21:33:35 etcd[25080]: sync duration of 4.802069156s, expected less than 1s
Mar 27 21:33:44 etcd[25080]: sync duration of 8.406615334s, expected less than 1s
Mar 27 21:33:54 etcd[25080]: got unexpected response error (etcdserver: request timed out)
Mar 27 21:33:56 etcd[25080]: got unexpected response error (etcdserver: request timed out) [merged 3 repeated lines in 1.8s]
Mar 27 21:33:57 etcd[25080]: sync duration of 12.481909793s, expected less than 1s
Mar 27 21:34:02 etcd[25080]: sync duration of 4.819143516s, expected less than 1s
```

@sdodson is there a hook on the installer that we can use to increase `etcd` verbosity? We're running with Origin `--loglevel=5`.

Other API requests do not seem to be affected at the same time. @smarterclayton does Ginkgo randomize the order of tests? Should we be looking to see what else is running at the same time? Why is it _this_ test that always has `etcd` fall over?

**Comment by miminar** [comment]
> does Ginkgo randomize the order of tests?

Looks like it. This time it was 193 out of 215 tests with different predecessors (compare with the [comment](https://github.com/openshift/origin/issues/13067#issuecomment-288095269) above):

```
191  [It] should *not* be restarted with a exec "cat /tmp/health" liveness probe [Conformance]
192  [It] should delete all failed deployer pods and hook pods
193  [Fail] [builds][Conformance] oc new-app [It] should succeed with a --name of 58 characters
```

**Comment by stevekuznetsov** [comment]
So do we expect this specific test to be triggering `etcd` to fall over? Post-mortem what sort of metrics would we need to debug this? We get configurable logging for `etcd` in https://github.com/openshift/openshift-ansible/pull/3782 after which I'll get more for us out of that service

**Comment by stevekuznetsov** [comment]
I think this is us failing to get `/var/lib/etcd` to be a tmpfs ... I can fix that

**Comment by kargakis** [comment]
https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/742/testReport/junit/(root)/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/

**Comment by smarterclayton** [comment]
This is indeed etcd failing over.

There is another problem.  We are using quorum reads from etcd, which
changed how errors are handled for authorization.  It's possible that we
should return 429 retry when etcd returns this error, even for auth.

On Apr 14, 2017, at 10:55 AM, Michail Kargakis <notifications@github.com>
wrote:

https://ci.openshift.redhat.com/jenkins/job/test_pull_request_origin/742/testReport/junit/(root)/Extended/_builds__Conformance__oc_new_app_should_succeed_with_a___name_of_58_characters/

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/issues/13067#issuecomment-294167946>,
or mute the thread

**Comment by ncdc** [comment]
@smarterclayton do you have an idea for how to attack fixing this?

**Comment by stevekuznetsov** [comment]
@ncdc I've been trying to tackle it in https://github.com/openshift/aos-cd-jobs/pull/169 but I keep seeing etcd just fall over, and setting `$ETCD_DEBUG` on has not really helped me to figure out why ...

**Comment by ncdc** [comment]
@stevekuznetsov have we been able to reproduce this in a VM that we control i.e. outside of CI?

**Comment by stevekuznetsov** [comment]
I've had very limited bandwidth so I have not been able to dig very deep into it

**Comment by stevekuznetsov** [comment]
Not a blocker, only affects test cases

**Comment by bparees** [comment]
p1 is used on flakes because of the impact they have on developer productivity.  we may need a more nuanced release process to make it clear the release is not blocked by them, but they shouldn't get relegated to p2 limbo if they are occurring frequently.

**Comment by stevekuznetsov** [comment]
Of course we need a better process, but for as long as we've been labeling flakes P1 we've done the demotion/promotion dance around their priority when they span a release ... the P1 is triggering blocker bug mechanisms that we don't need it to trigger as it does not block the release. Not really sure why we should treat _this_ flake differently from the rest.

**Comment by bparees** [comment]
That dance is news to me.  We have ignored them on blocker calls but i'm not aware of the process requiring us to formally mark them down to get through the release.  @pweil-

**Comment by bparees** [comment]
(and i'm not suggesting this flake be treated differently, it's just the only one i personally saw getting marked down when it's occurring frequently)

**Comment by bparees** [comment]
I've updated this flake to reflect the current root cause (which is cropping up in all sorts of test failures now): 'the server has asked for the client to provide credentials'

My understanding is we still think that is fundamentally caused by an etcd hiccup.

**Comment by stevekuznetsov** [comment]
Should be fixed in https://github.com/openshift/aos-cd-jobs/pull/199

**Comment by stevekuznetsov** [comment]
Haven't seen it since. Closing.
